hibikase/PlaybackWidget.cpp

```diff
@@ -100,6 +113,29 @@ void PlaybackWidget::OnPlayButtonClicked()
         QMetaObject::invokeMethod(m_worker, "Stop");
 }
 
+void PlaybackWidget::OnTimeSliderMoved(int value)
+{
+    if (!m_time_slider->isSliderDown())
+        return;
+
+    // Update karaoke text but not audio when dragging slider
+    // (It would also be possible to update audio, but you may not like the sound)
+    emit TimeUpdated(std::chrono::milliseconds(value));
+}
+
+void PlaybackWidget::OnTimeSliderReleased()
+{
+    if (!m_worker)
+        return;
+
+    // Audio stopped while seeking, restart it
+    if (m_state != QAudio::State::ActiveState)
+        QMetaObject::invokeMethod(m_worker, "Play");
```

Comment:
Could you move this check to inside the `Seek` function? That way there is no race condition. (Right now it takes a little while for `m_state` to update after the playback actually ends.)

Reply:
Fair! That I can do. Actually, I was wondering if I should just merge Play and Seek. The only difference is one starts implicitly at 0.

Reply:
That should probably be fine.

Reply:
Or, actually... We should keep in mind that we want to support seeking while paused, once pausing gets added.

Reply:
I'll split it apart again later in that case, the refactoring I did here is not wasted as I improved it in another way as well.